scylla-cql/src/types/serialize/serialize_row.rs

```diff
@@ -0,0 +1,44 @@
+use std::{any::Any, sync::Arc};
+
+use crate::{_macro_internal::ValueList, frame::response::result::ColumnSpec};
+
+pub struct RowSerializationContext<'a> {
+    columns: &'a [ColumnSpec],
+}
+
+impl<'a> RowSerializationContext<'a> {
+    pub fn column_by_idx(&self, i: usize) -> Option<&ColumnSpec> {
+        self.columns.get(i)
+    }
+
+    // TODO: change RowSerializationContext to make this faster
+    pub fn column_by_name(&self, target: &str) -> Option<&ColumnSpec> {
+        self.columns.iter().find(|&c| c.name == target)
+    }
+}
+
+pub trait SerializeRow {
+    fn preliminary_type_check(&self, ctx: &RowSerializationContext<'_>) -> bool;
+    fn serialize(
+        &self,
+        ctx: &RowSerializationContext<'_>,
+        out: &mut Vec<u8>,
+    ) -> Result<(), Arc<dyn Any + Send + Sync>>;
+}
+
+impl<T: ValueList> SerializeRow for T {
+    fn preliminary_type_check(&self, _ctx: &RowSerializationContext<'_>) -> bool {
+        true
+    }
+
+    fn serialize(
+        &self,
+        _ctx: &RowSerializationContext<'_>,
+        out: &mut Vec<u8>,
+    ) -> Result<(), Arc<dyn Any + Send + Sync>> {
```

Comment:
Please introduce an alias for `Arc<dyn Any + Send + Sync>`, e.g. `SerializationError`.